### PR TITLE
bugfix/#2156-fix-visibility-tags-after-preview

### PR DESCRIPTION
**Pull request by nazarhlyvka**
fixed-main-issue and fixed tests

### CODE REVIEW

File: src/app/main/component/eco-news/components/create-edit-news/create-edit-news.component.html [hunk 1]
```diff
@@ -27,6 +27,7 @@ <h3>{{ 'create-news.tags-title' | translate }}</h3>
                 (click)="addFilters(filter)"
                 type="button"
                 [ngClass]="{ 'filters-color': filter.isActive }"
+                class="tagNews"
```
Comment (dyadyushko) [inline src/app/main/component/eco-news/components/create-edit-news/create-edit-news.component.html:30]: We should use not the camel case in html classes/ids, but - sign to separate words

File: src/app/main/component/eco-news/components/create-edit-news/create-edit-news.component.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ export class CreateEditNewsComponent extends FormBaseComponent implements OnInit
   public month: number = new Date().getMonth();
   public author: string = localStorage.getItem('name');
   public attributes: ActionInterface;
-  public filters: Array<FilterModel>;
+  public filters: FilterModel[] | [] = [];
```
Comment (dyadyushko) [inline src/app/main/component/eco-news/components/create-edit-news/create-edit-news.component.ts:37]: Do we need second type of regular array? Empty array value still acceptable as FilterModel[] type

**Review by rostikdoroshenko — APPROVED** [review]

File: src/app/main/component/eco-news/components/create-edit-news/create-edit-news.component.html (hunk 1 above)
Comment (nazarhlyvka) [reply src/app/main/component/eco-news/components/create-edit-news/create-edit-news.component.html]: done

File: src/app/main/component/eco-news/components/create-edit-news/create-edit-news.component.ts (hunk 2 above)
Comment (nazarhlyvka) [reply src/app/main/component/eco-news/components/create-edit-news/create-edit-news.component.ts]: done

File: src/app/main/component/eco-news/components/create-edit-news/create-edit-news.component.ts [hunk 3]
```diff
@@ -104,6 +104,11 @@ export class CreateEditNewsComponent extends FormBaseComponent implements OnInit
   }
 
   private getAllTags() {
+    if (localStorage.getItem('newsTags')) {
```
Comment (JimHolman1199) [inline src/app/main/component/eco-news/components/create-edit-news/create-edit-news.component.ts:107]: we should use localStorageSerrvice for this

File: src/app/main/component/eco-news/components/create-edit-news/create-edit-news.component.ts [hunk 4]
```diff
@@ -191,6 +196,8 @@ export class CreateEditNewsComponent extends FormBaseComponent implements OnInit
         })
       )
       .subscribe(() => this.escapeFromCreatePage());
+
+    localStorage.removeItem('newsTags');
```
Comment (JimHolman1199) [inline src/app/main/component/eco-news/components/create-edit-news/create-edit-news.component.ts:200]: we should use localStorageSerrvice for this

File: src/app/main/component/eco-news/components/create-edit-news/create-edit-news.component.ts [hunk 5]
```diff
@@ -278,7 +285,9 @@ export class CreateEditNewsComponent extends FormBaseComponent implements OnInit
 
   public toggleIsActive(filterObj: FilterModel, newValue: boolean): void {
     const index = this.filters.findIndex((item: FilterModel) => item.name === filterObj.name);
-    this.filters = this.filterArr({ name: filterObj.name, isActive: newValue }, index);
+    const changedtags = this.filterArr({ name: filterObj.name, isActive: newValue }, index);
+    this.filters = changedtags;
+    localStorage.setItem('newsTags', JSON.stringify(changedtags));
```
Comment (JimHolman1199) [inline src/app/main/component/eco-news/components/create-edit-news/create-edit-news.component.ts:290]: we should use localStorageSerrvice for this

**Review by dyadyushko — APPROVED** [review]

File: src/app/main/component/eco-news/components/create-edit-news/create-edit-news.component.ts (hunk 5 above)
Comment (nazarhlyvka) [reply src/app/main/component/eco-news/components/create-edit-news/create-edit-news.component.ts]: done

File: src/app/main/component/eco-news/components/create-edit-news/create-edit-news.component.ts (hunk 4 above)
Comment (nazarhlyvka) [reply src/app/main/component/eco-news/components/create-edit-news/create-edit-news.component.ts]: done

File: src/app/main/component/eco-news/components/create-edit-news/create-edit-news.component.ts (hunk 3 above)
Comment (nazarhlyvka) [reply src/app/main/component/eco-news/components/create-edit-news/create-edit-news.component.ts]: done

**Review by s1auka — APPROVED** [review]

**Review by andkuzma — APPROVED** [review]

**Review by JimHolman1199 — APPROVED** [review]

**Pull request merged**